tests/publiccloud/run_ltp.pm

```diff
@@ -152,42 +195,50 @@ sub run {
     } else {
         record_info("Exclude", "None");
     }
+    return $skip_tests;
+}
 
+sub prepare_kirk {
+    my ($self, $instance) = @_;
     my $kirk_repo = get_var("LTP_RUN_NG_REPO", "https://github.com/linux-test-project/kirk.git");
     my $kirk_branch = get_var("LTP_RUN_NG_BRANCH", "master");
     record_info('LTP RUNNER REPO', "Repo: " . $kirk_repo . "\nBranch: " . $kirk_branch);
     script_retry("git clone -q --single-branch -b $kirk_branch --depth 1 $kirk_repo", retry => 5, delay => 60, timeout => 300);
     $instance->run_ssh_command(cmd => 'sudo CREATE_ENTRIES=1 ' . get_ltproot() . '/IDcheck.sh', timeout => 300);
     record_info('Kernel info', $instance->run_ssh_command(cmd => q(rpm -qa 'kernel*' --qf '%{NAME}\n' | sort | uniq | xargs rpm -qi)));
     if (get_var('PUBLIC_CLOUD_INSTANCE_TYPE') =~ /-metal$/) {
-        # The metal detector fails on GCP because it may return "google"
         record_info('VM type', $instance->run_ssh_command(cmd => '! systemd-detect-virt')) unless is_gce;
     } else {
         record_info('VM type', $instance->run_ssh_command(cmd => 'systemd-detect-virt'));
     }
+    assert_script_run("cd kirk");
+    my $ghash = script_output("git rev-parse HEAD", proceed_on_failure => 1);
+    set_var("LTP_RUN_NG_GIT_HASH", $ghash);
+    record_info("KIRK_GIT_HASH", "$ghash");
+    assert_script_run("python3.11 -m venv env311");
+    assert_script_run("source env311/bin/activate");
```

Comment:
I would use a generic name for the directory that doesn't depend in the python version so we can just update the python3.11 commands.

```suggestion
    assert_script_run("python3.11 -m venv venv");
    assert_script_run("source venv/bin/activate");
```